### DISCUSSION TITLE
domain purchased through GoDaddy does not return the DNS zones in the response, getting a null value instead

### DISCUSSION THREAD

**Discussion by max2-wms** [polymind-inc/acmebot#761] [Q&A]
Describe the bug
I have a domain with GoDaddy, I filled out all the environment variables in the configuration of the Function app including the GoDaddy DNS provider values. but when I go to the dashboard and try to get it to generate the SSL certificate, it does not return the list of DNS zones, the value returned is actually null for some reason. the dropdown menu in the popup says "fetch error". I contacted GoDaddy customer support and they provided this answer. is there anything I can do to bypass the limitation, what would be the best approach?

Thanks

https://github.com/user-attachments/assets/fbfc2ce0-50ec-4404-9efe-2a1e6c7a3dd3

![Screenshot 2024-09-13 164352](https://github.com/user-attachments/assets/1fbb86e3-d8d1-4b29-841c-9b57ac7c3bb2)
![Screenshot 2024-09-13 165353](https://github.com/user-attachments/assets/bd1d5d25-2214-4ccf-9fbb-9f6e35eaf7c3)
![Screenshot 2024-09-13 165412](https://github.com/user-attachments/assets/49d4be72-561e-4afb-a836-6865e57de017)
![Screenshot 2024-09-16 151955](https://github.com/user-attachments/assets/d4f58ea5-14d3-445f-8158-a50d7f97b54b)

**Comment by JohnLBevan** [comment]
Sadly this is an issue with GoDaddy's API - there's no solution as a client of the API to bypass their rules as the provider of that API.

There are workarounds that don't rely on GoDaddy's API though (previously discussed on [#722](https://github.com/shibayan/keyvault-acmebot/discussions/722)).

- Use a different DNS provider (name server).  Specifically; even if you use GoDaddy to purchase your domain, you don't have to continue to use them to manage your DNS records.  You can setup zones for these domains elsewhere (e.g. Azure/AWS) then update GoDaddy to use the NS records of these hosted zones instead of its own (see [edit-my-domain-nameservers](https://www.godaddy.com/help/edit-my-domain-nameservers-664)).
- Though not an option today, a new feature to handle CNames ([#159](https://github.com/shibayan/keyvault-acmebot/issues/159)) would similarly allow a workaround - for that you'd still need a DNS zone managed by a different provider (e.g. Azure/AWS), but you could keep your main domain in GoDaddy and just point its acme validation record at a record in your non-GoDaddy domain, which AcmeBot can then manage.  This is typically a good solution for special cases where you have a specific reason to keep your domain's name servers wtih the unsupported provider; but if you have the freedom to repoint your name servers, the previous solution's better as it's supported today, is simpler to setup, and you get the benefit of being able to mange other records via the API (e.g. if kubernetes creates DNS entries for your public ingress rules, or you want to use Terraform to manage your records in IaC, etc).
